### ISSUE TITLE
how to predict on new image

### ISSUE THREAD

**Issue by kbrajwani** [microsoft/unilm#374]
i am using layoutlmv2 model.
let's say i have image and it's annotation file and i want to predict on only that image. so how can i predict that. Is their any script is built already for inference.
I have experience with layoutlmv1 model.

**Comment by sherlocked27** [comment]
You need to first tokenize and add padding  [here](https://github.com/microsoft/unilm/blob/00a147f4ce71c0b21589c07ef8ffca406c12e5be/layoutlmft/examples/run_funsd.py#L165) then get prediction from the model [here](https://github.com/microsoft/unilm/blob/00a147f4ce71c0b21589c07ef8ffca406c12e5be/layoutlmft/examples/run_funsd.py#L338).

You can create your own eval function that does the same, after padding and tokenization
`          input_ids=batch['input_ids'].to(device)
            bbox=batch['bbox'].to(device)
            attention_mask=batch['attention_mask'].to(device)
            token_type_ids=batch['token_type_ids'].to(device)
            labels=batch['labels'].to(device)
            resized_images = batch['resized_image'].to(device) 
            resized_and_aligned_bounding_boxes = batch['resized_and_aligned_bounding_boxes'].to(device)
            outputs = model(image = resized_images,input_ids=input_ids, bbox=bbox, attention_mask=attention_mask, token_type_ids=token_type_ids,labels=labels)
            tmp_eval_loss = outputs.loss
            logits = outputs.logits
            eval_loss += tmp_eval_loss.item()
            nb_eval_steps += 1
            if preds is None:
                preds = logits.detach().cpu().numpy()
                out_label_ids = labels.detach().cpu().numpy()
            else:
                preds = np.append(preds, logits.detach().cpu().numpy(), axis=0)
                out_label_ids = np.append(
                    out_label_ids, labels.detach().cpu().numpy(), axis=0
                )`

**Comment by kbrajwani** [comment]
hey i tweak the dataloader and able to get label and prediction on token level. I have already written a script for v1 so it would be great if i can able to get test_image.txt file as created in v1 and prediction logits on words level.

**Comment by sherlocked27** [comment]
Since you already have the prediction at token_id level, you can try to create the input_id to word by using `tokenizer.convert_ids_to_tokens(input_id)`. 
You may need to join the words back as tokenizer split the words into multiple parts.

**Comment by kbrajwani** [comment]
Thanks @sherlocked27  thanks for support. let me try this solutions.

**Comment by kbrajwani** [comment]
I tried the way you tell me but still getting error. can you please look this image and suggest something.
![image](https://user-images.githubusercontent.com/29722986/127991089-3355b0c2-bcb1-43ee-b2dc-7412df9266d4.png)

**Comment by sherlocked27** [comment]
Removing .tensor if it is already a Tensor solves the error but it is a workaround. 
```
 images_input = (
            (images if torch.is_tensor(images) else images.tensor) - self.pixel_mean
        ) 
```
You can check [this](https://github.com/microsoft/unilm/blob/00a147f4ce71c0b21589c07ef8ffca406c12e5be/layoutlmft/layoutlmft/data/datasets/funsd.py#L62) also, haven't tried initialising like this but it may work, as it should be ImageList but some initialisation is causing issue and it is a Tensor
